File: lib/kotlin-mpp/src/commonMain/kotlin/com/sdercolin/utaformatix/data/Track.kt
package com.sdercolin.utaformatix.data

import kotlinx.serialization.Serializable

/**
 * Track model
 * @param name Track name
 * @param notes Notes in the track
 * @param pitch Pitch data bound to the track (if any)
 */
@Serializable
data class Track(
    val name: String,
    val notes: List<Note>,
    val pitch: Pitch?
)

File: lib/typescript/src/lib/track.ts
import { Note } from './note';
import { Pitch } from './pitch';

/**
 * Track model.
 * @param name - Track name.
 * @param notes - Notes in the track.
 * @param pitch - Pitch data bound to the track (if any).
 */
export type Track = {
  name: string;
  notes: Note[];
  pitch?: Pitch;
};
